app/src/config/logger.ts

```diff
@@ -1,5 +1,36 @@
 import {get} from '../utils/env';
+import {ConsoleTransportOptions} from 'winston/lib/winston/transports';
+import {DiscordTransportOptions} from '../logger/DiscordTransport';
+
+export interface TransportsIndex {
+    console: ConsoleTransportOptions;
+    discord: DiscordTransportOptions;
+}
+
+export type TransportConfig<K extends keyof TransportsIndex = keyof TransportsIndex> = {
+    name: K;
+    options: TransportsIndex[K];
+} & ({ whitelist?: Record<string, boolean> } | { blacklist?: Record<string, boolean> });
 
 export default class Logger {
     public readonly level = get('LOG_LEVEL', get('NODE_ENV', 'development') === 'development' ? 'debug' : 'info');
+
+    public readonly globalFilter: Record<string, boolean> = {};
```

Comment:
Could you add an example? Also are you sure this will still allow errors to come through but not info / debug messages?

Reply:
Basically for ever logger you use `getLogger` which you supply with a label. In that you can either black or whitelist these labels. For configurability you can set them to true to include them(so you can overwrite them with environment variables).

Reply:
Ok, but would this still suppress everything? We only want certain log levels to be suppressed, errors always being unsuppressed.

Reply:
Ooh, shoot. Didn't think about that.

Reply:
Ooh, yeah didn't think that.

Reply:
Done